### ISSUE TITLE
Parsing Expressions

### ISSUE THREAD

**Issue by colms** [munificent/craftinginterpreters#76]
#### potential changes

- "Likewise, some operators are non-associative. That means it’s an error to use that operator more than once in a sequence." - I wanted to know which operators.
- "Operators with *higher* precedence are evaluated before operators with lower precedence." - this sentence just felt unbalanced to me because *higher* was italicized but lower wasn't. This could just be an anal-retentive thing with me.
- "left-associative" is hyphenated but "left recursive" isn't. I don't know which convention to stick with.
  - It might be worth grepping to find words that appear in both hyphenated and and unhyphenated form. Heck, also no hyphen, no space form. I feel there may be a few throughout the book.
  - "lowest-precedence" has hyphen, later "lowest precedence" doesn't.
  - "highest precedence" no hyphen, "higher-precedence" has.
- "The JavaScript" - casing of T
- "transliteration" - I don't know what that means.
- "The function for a left recursive rule would immediately call itself, which would call itself again, and so on, until the parser hit a stack overflow and died." - some tense issue in this sentence makes it feel weird.
- "if don't see" - if we don't see
- These two:
  - "This returns true ..." - my concern is that a new page could occur here in print and the reader would have to page back to be sure what "This" is because it's a code block. (I just verbified "page" ... and verb.) I might go with "This `check()` method returns true"
  - I'd do the same for the sister function - "This consumes" => "This `advance()` method consumes"
- starting a sentence with a function or variable name e.g. `isAtEnd()` and `peek()`, is sometimes considered bad form. Personally I think it's fine but just making you aware.
- "Finally it" - maybe "Finally, it"
- "virtually identical" - to what? Maybe add "to the ..."
- "now `term()`. " - "now `term()` instead of `comparison()`" (I think that's the counterpart it's referring to)
- "how to parse lumps" - I read that "lumps" as a noun thinking it was a new term before going on, realizing my stupidity and rereading the sentence. Maybe "how to parse puts". Interestingly though, a garden path sentence would be well placed when talking about predictive parsers.
- "entrypoint" - I think this is correctly one word but I just want to be sure.
- "One way to" - I'd rephrase this sentence. I get what it's saying but there's something wrong here.

**Comment by munificent** [comment]
> "Operators with higher precedence are evaluated before operators with lower precedence." - this sentence just felt unbalanced to me because higher was italicized but lower wasn't. This could just be an anal-retentive thing with me.

Agreed. It was asymmetric.

> "left-associative" is hyphenated but "left recursive" isn't. I don't know which convention to stick with.

Started tracking this in the style guide. Going off some Googling and Wikipedia, "left-associative" and "left-recursive" are always hyphenated. "Left recursion" isn't. "First-class" is hyphenated when preceding a noun. English! 🙃 

> It might be worth grepping to find words that appear in both hyphenated and and unhyphenated form. Heck, also no hyphen, no space form. I feel there may be a few throughout the book.

I'm sure there are. I try to be consistent, but it's hard. When the book is done, I'll probably do a proofreading pass after getting my hands on a copy of the Chicago Manual of Style.

> "transliteration" - I don't know what that means.

Apparently I didn't either. Fixed.

> These two:
> "This returns true ..." - my concern is that a new page could occur here in print and the reader would have to page back to be sure what "This" is because it's a code block. (I just verbified "page" ... and verb.) I might go with "This check() method returns true"

For bigger code blocks, I'm inclined to do that, but for this one it feels redundant to me to mention the method name when it's right there a few lines up.

When I do the print edition, I'll manually typeset it and control exactly where every page break is. :)

> starting a sentence with a function or variable name e.g. isAtEnd() and peek(), is sometimes considered bad form. Personally I think it's fine but just making you aware.

Yeah, I try to avoid it, but it felt awkward to do so here.

> "entrypoint" - I think this is correctly one word but I just want to be sure.

Hmm, a handful of dictionaries don't think it's one word or two. Reworded.

> "One way to" - I'd rephrase this sentence. I get what it's saying but there's something wrong here.

Rewrote the whole section.

All the other stuff is done. Thanks!
